### PR TITLE
feat: add GithubInstallationAuthenticator auth option

### PR DESCRIPTION
**Pull request by henryl**
## What
Installation tokens expire after 1hr meaning in a typical sync, any token that we pass to the connection will expire before the sync is complete. This PR modifies the source-github connector to generate its own installation tokens.

## How
We use an [Evervault function](https://github.com/Attuned-Corp/evervault-github-installation-token-function) to generate and refresh installation access tokens.

Passing `credentials.option_title = "Installation ID"` during connection configuration selects into this auth mode, which will require the following `credential.{installation_id, app_id, app_private_key, evervault_key}` keys. The `app_private_key` should be EV encrypted.

### CODE REVIEW

**Comment by henryl** [comment]
Should we be using airbyte secrets for the `GITHUB_APP_PRIVATE_KEY` and `EVERVAULT_API_KEY`? It is poorly documented and I want to discuss.

**Comment by ncurbez** [comment]
the docs mention using GCP/AWS to store secrets, but that seems like using GCP/AWS to store all credentials in general. We could make these secrets part of the authentication configuration for the connector, similar to how it stores a token now. Maybe if we have a separate Evervault API key for this use case, we can treat it as a long lived token and pass it that way?

File: airbyte-integrations/connectors/source-github/source_github/authenticator.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        self._installation_id = installation_id
+        self._token = None
+        self._token_expires_at = None
+
+    def _ensure_token(self):
+        if not self._token or time.time() > self._token_expires_at:
+            ev_res = evervault.run('github-installation-token-function', {
+                'appId': self._app_id,
+                'installationId': self._installation_id,
+                'privateKEy': self._private_key,
```
Comment (ncurbez) [inline airbyte-integrations/connectors/source-github/source_github/authenticator.py:42]: capitalization caught my eye here, is it a typo?

Comment (henryl) [reply airbyte-integrations/connectors/source-github/source_github/authenticator.py:42]: Ah - yep haven't tried running the sync yet. Good catch.

**Comment by henryl** [comment]
cc: @ncurbez I think this works now. I'm going to merge for now but create an issue for us to discuss how to connect the Docker image into our infra.

**Pull request merged**